rb_ws/src/buggy/scripts/2d_sim/engine.py

```diff
@@ -87,15 +94,20 @@ def update_steering_angle(self, data: Float64):
         with self.lock:
             self.steering_angle = data.data
 
-    def update_velocity(self, data: Float64):
+    def update_velocity(self, data: Float64, source):
         """Updates the velocity as callback function for subscriber
 
         Args:
             data (Float64): velocity in m/s
+            source (string): whether incoming data is manual or simulated
         """
-        with self.lock:
-            self.velocity = data.data
-
+        if (source == "simulated" and not Simulator.MANUAL_V_FLAG):
```

Comment:
Why do we need to check these two variables? They seem redundant. Removing this will probably resolve the "else" comment I made. But removing it will probably require only having one velocity_subscriber at a time.

Reply:
Seconded, see my review